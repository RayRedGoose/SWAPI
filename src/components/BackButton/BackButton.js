import React, { Component } from 'react'
import { Redirect } from 'react-router-dom'
import './BackButton.scss'

export default class BackButton extends Component {
  constructor() {
    super()
    this.state = {
      isRedirected: false
    }
  }

  redirect = () => {
    this.setState({isRedirected: true})
  }

  render() {
    return (
      (this.state.isRedirected)
        ? <Redirect to='/movies' />
        : <button
          className="back-btn"
          onClick={this.redirect}>back to all movies</button>
    )
  }
}
