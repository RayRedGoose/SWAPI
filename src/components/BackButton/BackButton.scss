.back-btn {
  @extend %background-images;
  background-color: rgba(255, 255, 255, 0.9);
  border: none;
  color: #5A2C05;
  cursor: pointer;
  font-size: 18px;
  text-transform: lowercase;
  padding: 0.4em 2em;
  position: absolute;
  top: 20px; left: 40px;
  &:hover {
    @extend %background-images;
    background: #5A2C05;
    color: #F1AA61;
  }
}

%background-images {
  background-image: url('../../images/back.svg');
  background-size: 20px;
  background-position: 10px center;
  background-repeat: no-repeat;
}
