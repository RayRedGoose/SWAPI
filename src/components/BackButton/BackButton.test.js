import React from 'react'
import { shallow } from 'enzyme'
import BackButton from './BackButton'

describe("BackButton", () => {
  let button

  beforeEach(() => {
    button = shallow(<BackButton />)
  })

  it("should match snapshot if isRedirected state is false", () => {
    expect(button).toMatchSnapshot()
  })

  it("should match snapshot if isRedirected state is true", () => {
    button.setState({isRedirected: true})

    expect(button).toMatchSnapshot()
  })

  it("should call redirect when button is clicked", () => {
    const spy = jest.spyOn(button.instance(), 'redirect').mockImplementation(arg => arg)
    button.instance().forceUpdate()

    button.find('button').simulate('click')

    expect(spy).toHaveBeenCalled()
  })

  it("should change isRedirected state when redirect is called", () => {
    expect(button.state('isRedirected')).toEqual(false)

    button.instance().redirect()

    expect(button.state('isRedirected')).toEqual(true)
  })
})
