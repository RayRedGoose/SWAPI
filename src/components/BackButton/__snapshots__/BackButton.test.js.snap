// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton should match snapshot if isRedirected state is false 1`] = `
<button
  className="back-btn"
  onClick={[Function]}
>
  back to all movies
</button>
`;

exports[`BackButton should match snapshot if isRedirected state is true 1`] = `
<Redirect
  to="/movies"
/>
`;
